lib/xcodeproj/project/object/native_target.rb

```diff
@@ -552,6 +552,36 @@ def add_resources(resource_file_references)
           end
         end
 
+        # remove on demand resource files to the resources build phase of the target.
+        #
+        # @param  [Array<PBXFileReference>] on_demand_resource_file_references
+        #         the files references of the on demand resources to the target.
+        #
+        # @return [void]
+        #
+        def remove_on_demand_resources(on_demand_resource_file_references)
+          on_demand_resource_file_references.each do |file|
+            resources_build_phase.remove_file_reference(file)
+          end
+        end
+
+        # Adds on demand resource files to the resources build phase of the target.
+        #
+        # @param  [Array<PBXFileReference>] on_demand_resource_file_references
+        #         the files references of the on demand resources to the target.
+        #
+        # @return [void]
+        #
+        def add_on_demand_resources(on_demand_resource_file_references)
+          on_demand_resource_file_references.each do |file|
+            next if resources_build_phase.include?(file)
+            build_file = project.new(PBXBuildFile)
+            build_file.file_ref = file
+            build_file.settings = (build_file.settings ||= {}).merge({"ASSET_TAGS" => ["OnDemand"]})
```

Comment:
I don't think we want to be hard-coding asset tags here. According to [Apple's docs](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/On_Demand_Resources_Guide/Tagging.html#//apple_ref/doc/uid/TP40015083-CH3-SW1) tags are supposed to be manually specified, and those tags are then used in code to [download the assets](https://developer.apple.com/documentation/foundation/nsbundleresourcerequest)

Reply:
thanks, now use the podspec on_demand_resources specified tag 
https://github.com/CocoaPods/Xcodeproj/pull/742/commits/81403b6248a44e0ad15c151680cb22360d06141c#diff-5ee67a82fd5ae8aa882ddaf26476e138R577